### PR TITLE
Added Rate Limiter in resilience

### PR DESCRIPTION
**Pull request by CharlesDuboisSAP**
## What Has Changed?

Rate Limiter has been added in resilience capabilities

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- ~[ ] You verified all new and changed links still work (changing the `id` or name of a file can break links)~
- ~[ ] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.~

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "$schema": "http://json.schemastore.org/prettierrc",
   "singleQuote": true,
   "arrowParens": "avoid",
-  "trailingComma": "none"
+  "trailingComma": "none",
+  "endOfLine": "crlf"
```
Comment (newtork) [inline .prettierrc:6]: Why is that chang?
```suggestion
  "trailingComma": "none"
```

File: docs/java/features/resilience/resilience.mdx [hunk 2]
```diff
@@ -244,6 +244,17 @@ The following resilience patterns are available and can be configured in a Resil
         Limit how long an operation may run before it should be interrupted
       </td>
     </tr>
+    <tr>
+      <td>Rate Limiter</td>
+      <td>
+        <em>
+          <a href="https://help.sap.com/http.svc/rc/b579bf8578954412aea2b458e8452201/1.0/en-US/com/sap/cloud/sdk/cloudplatform/resilience/ResilienceConfiguration.RateLimiterConfiguration.html">
+            RateLimiterConfiguration
```
Comment (Johannes-Schneider) [inline docs/java/features/resilience/resilience.mdx:252]: Unfortunately, this link doesn't work as long as we are not uploading new Java Docs ... 😕

File: docs/java/features/resilience/resilience.mdx [hunk 3]
```diff
@@ -297,9 +304,10 @@ They are applied in the following order:
 
 1. Timeouts
```
Comment (KavithaSiva) [inline docs/java/features/resilience/resilience.mdx:305]: The order is mostly based upon the order in which the decorators are added in `Resilince4jDecorationStrategy`. Please check [here](https://github.wdf.sap.corp/MA/sdk/blob/develop/frameworks/resilience4j/src/main/java/com/sap/cloud/sdk/frameworks/resilience4j/Resilience4jDecorationStrategy.java#L58)
I also verified that although `BulkHead` is the first among resilience patterns a Timeout exception is still raised if applicable. Refer to this [extended test](https://github.wdf.sap.corp/MA/sdk/pull/7109)

Comment (KavithaSiva) [reply docs/java/features/resilience/resilience.mdx:305]: Just for reference, the initial order proposed by @MatKuhr and @CharlesDuboisSAP in the PR was:
1. Retries
2. Fallbacks
3. Circuit Breaker
4. Rate Limiter
5. Timeout
6. Bulkhead

But I changed the order to reflect that `Timeout` if applicable would always be called first. 
@MatKuhr, @CharlesDuboisSAP  Can you please let me know on why you think the order must be changed.

File: docs/java/features/resilience/resilience.mdx [hunk 4]
```diff
@@ -295,12 +302,13 @@ You can find good explanations on how the individual patterns behave on the [doc
...
 
-1. Timeouts
-2. Bulkhead
-3. Circuit Breaker
-4. Retries
-5. Fallbacks
+Fallback( Retry ( CircuitBreaker ( RateLimiter ( TimeLimiter ( Bulkhead ( Function ) ) ) ) ) )
+
+If you read from right to left, it shows you the order in which the aspects will be applied i.e `Fallbacks` are called last while `Bulkhead` is the first aspect applied.
+Hence,exceptions are also propagated from right to left.
```
Comment (Johannes-Schneider) [inline docs/java/features/resilience/resilience.mdx:308]:
```suggestion
If you read from right to left, it shows you the order in which the aspects will be applied.
For example, `Fallbacks` are called last while `Bulkhead` is the first aspect applied.
Hence, exceptions are also propagated from right to left.
```

File: docs/java/features/resilience/resilience.mdx [hunk 5]
```diff
@@ -295,12 +302,13 @@ You can find good explanations on how the individual patterns behave on the [doc
...
+
+Based on the order, the following inferences(not exhaustive) can be made:
```
Comment (Johannes-Schneider) [inline docs/java/features/resilience/resilience.mdx:310]:
```suggestion
Based on the order, the following inferences (not exhaustive) can be made:
```

File: docs/java/features/resilience/resilience.mdx [hunk 6]
```diff
@@ -295,12 +302,13 @@ You can find good explanations on how the individual patterns behave on the [doc
 Be aware that the patterns interact with each other.
 They are applied in the following order:
 
-1. Timeouts
-2. Bulkhead
-3. Circuit Breaker
-4. Retries
-5. Fallbacks
+Fallback( Retry ( CircuitBreaker ( RateLimiter ( TimeLimiter ( Bulkhead ( Function ) ) ) ) ) )
```
Comment (Johannes-Schneider) [inline docs/java/features/resilience/resilience.mdx:305]:
```suggestion
Fallback ( Retry ( CircuitBreaker ( RateLimiter ( TimeLimiter ( Bulkhead ( Function ) ) ) ) ) )
```

**Review by Johannes-Schneider — COMMENTED** [review]
This update makes sense to me, especially since we cross-checked it with the Resilience4J documentation.

Nevertheless, I'm wondering: Have you been able to (partially) confirm this order with the tests you wrote?

---

I also left some cosmetic suggestions.

**Comment by KavithaSiva** [comment]
> This update makes sense to me, especially since we cross-checked it with the Resilience4J documentation.
> 
> Nevertheless, I'm wondering: Have you been able to (partially) confirm this order with the tests you wrote?
> 
> I also left some cosmetic suggestions.

We aligned on introducing a test that checks the interaction between `Timeout` and  `BulkHead`

**Review by Johannes-Schneider — APPROVED** [review]
LGTM

**Pull request merged**